client/lib/site/index.js

```diff
@@ -61,14 +61,17 @@ Site.prototype.attributes = function( attributes ) {
 };
 
 Site.prototype.set = function( attributes ) {
-	var changed = false;
-
-	for ( var prop in attributes ) {
-		if ( attributes.hasOwnProperty( prop ) && ! isEqual( attributes[ prop ], this[ prop ] ) ) {
-			if ( undefined === attributes[ prop ] ) {
+	let changed = false;
```

Comment:
It's magic, but it works with the tests provided. I can't think of anything better so I think it is good to go 👍